spell/spell-reviewer-mainnet-checklist.md

```diff
@@ -170,7 +190,31 @@ Spell Actions:
     * [ ] Recipient Addresses match Doc
     * [ ] Payment Amounts match Doc
     * [ ] Follow Previous Patterns
+  * [ ] `Yank`
+    * [ ] Stream ID is matches exec
+    * [ ] `MCD_VEST_DAI` for DAI stream `yank`
+    * [ ] `MCD_VEST_MKR_TREASURY` for MKR stream `yank`
   * [ ] Ensure Recipient Addresses match `addresses_wallets.sol`
+* [ ] SubDAO Content
+  * [ ] SubDAO SubProxy spell execution
+    * [ ] SubDAO spell address matches exec doc
+    * [ ] SubDAO spell deployer is in `addresses_deployers`
+    * [ ] Executed using `ProxyLike(SUBDAO_PROXY).exec(SUBDAO_SPELL, abi.encodeWithSignature("execute()"));`
+    * [ ] Execution is NOT delegate call
+    * [ ] Gas cost will not be too high (low level call gas estimation is not done by our scripts)
+  * [ ] Maker Core (main spell) SubDAO actions (i.e. operate in Pause Proxy `DelegateCall` context)
+    * [ ] No SubDAO contract being interacted with is authed on a core contract like vat, etc. (script this eventually)
+    * [ ] SubDAO contract licensing and optimizations do not matter (not strictly)
+    * [ ] SubDAO contracts and all libraries / dependencies are verified (Blocking if not true)
+    * [ ] Upgradable SubDAO contracts
+      * [ ] Any upgradable contracts have the `PAUSE_PROXY` as their `admin` (i.e. the party that can upgrade)
+        * [ ] Any upgradable SubDAO contracts with an `admin` that is not `PAUSE_PROXY` are not authed on any core contracts (Blocking)
+    * [ ] All SubDAO content addresses (i.e. provided contract addresses or EOAs) present in the Maker Core spell are present in the exec doc and are correct. SubDAO addresses being authed or given any permissions MUST be in the exec doc. SubDAO addresses being called must be confirmed by the SubDAO spell team.
+      * [ ] IF addresses not PR'ed in by the SubDAO team (use git blame for example), SubDAO content addresses all have inline comment for provenance or source being OKed by SubDAO
+    * [ ] SubDAO actions match exec doc (only where inline with main spell code) and do not affect core contracts
```

Comment:
Let's clearly define two different terms for code in the core spell related to SubDAO (eg `SubDAO-related actions`) and code inside SubDAO spell (eg `actions inside SubDAO spell`) and then use them exclusively in the doc. Otherwise it's kind of a mess. 

From the current wording I don't get what you mean by `SubDAO actions` – SubDAO-related actions in the core spell are already covered in the bullet-points above, actions inside SubDAO spell are reviewed separately

Reply:
I agree, I will think about this more.

 SubDAO actions --> Subdao spell actions 
 SubDAO core actions --> Subdao requested changes to Maker Protocol contracts

Any feedback here?

Reply:
Also, do you want to clearly state where responsibility ends with the SubDAO-spell reviews? Eg: `SubDAO-spell content has at least two public approvals and is linked in the spell`

Reply:
> Also, do you want to clearly state where responsibility ends with the SubDAO-spell reviews? Eg: `SubDAO-spell content has at least two public approvals and is linked in the spell`

Not in the checklist yet since this is still up in the air and needs to be formalized - see my reply to the earlier point on this.

I agree that this definition should be in the checklist in principle but the checklist isn't a source of truth for things like this, so maybe let's regroup and settle it with stakeholder (subdao) input separately?

> The wording is not very usable cause it's quite long. Maybe introducing subspell/SubDAO-spell term would help? Ie `SubDAO-spell actions` for something that SubDAO writes and `SubDAO-related actions` for something that is contained within the core spell? If you agree, please edit the wording of the checklist

SubDAO actions == "SubDAO-spell actions"
SubDAO core / SubDAO core actions == "SubDAO-related actions"

Here, the wording needs to be considered carefully since again this isn't the source of truth on this stuff. Soemthing can be subdao related but also not maker core and also not in a spell (it may not even go through the pause proxy), so I limited it to 'core' which only the PP can execute and then 'actions' which is just "!core" i.e. subdao spell and other subdao related stuff.

Limiting spell team coverage to subdao core rather than "subdao related" limits the scope of stuff being covered by the core spell team in a better way IMO and is a more strict definition ('core' is clearer than 'related').

Open to feedback (or alternative names!) here :)